src/Uno.UI/UI/Xaml/Controls/ContentControl/ContentControl.cs

```diff
@@ -67,9 +67,19 @@ public virtual object Content
 		{
 			get
 			{
-				return this.IsDependencyPropertySet(ContentProperty)
-					? GetValue(ContentProperty)
-					: DataContext;
+				if(this.IsDependencyPropertySet(ContentProperty))
+				{
+					return GetValue(ContentProperty);
+				}
+				else if(ContentTemplate != null)
+				{										 
+				   return DataContext;
+				}
+				else
+				{
+					return null;
+				}
+
```

Comment:
Please add comment explaining each scenario.

Reply:
Also, I'm not sure if the original version was doing it properly as well. `GetValue(ContentControl.ContentProperty)` would return an incorrect value (e.g. in template binding scenarios).

We may need to change the local value instead, reacting do `DataContext`, `ContentTemplate` and `Content` changes.

Also, there are no tests for this scenario, could you add some? They can be unit tests, not UI tests.